Week2/MAKEME.md

```diff
@@ -26,23 +26,23 @@ This week we'll practice some more with writing SQL queries using JavaScript. Fo
 
 **Exercise 1: Keys**
 
-1. Create a table, called `Authors`. Give it the following fields: `(author_no(Primary Key), author_name, university, date_of_birth, h_index, gender)`
-2. Write a query that adds a `foreign key` column to `Authors` table that references the column `author_no`.
-Call this column `Collaborator`.
+1. Create a table, called `authors`. Give it the following fields: `(author_no(Primary Key), author_name, university, date_of_birth, h_index, gender)`
+2. Write a query that adds a column called `collaborator` to `authors` table that references the column `author_no`.
+For integrity add a `foreign key` on this column.
```

Comment:
They are supposed to deduce that.

Reply:
I'm not a fan of the deduction-type-exercises and some students didn't know what to make of it. 
But I do concur that it's not an improvement to make it this easy.
I think the sweet spot is to explain the problem case and let them come up with the solution. 

Something like 
> 2. Authors usually help out other authors with their papers. In that role they are called a collaborator. 
Add a column to `authors` to store the value that references another author and add a key for integrity validation.

What do you think about this version?

Reply:
I think `add a key for integrity validation` is extra information because I think you are meaning referential integrity here which is implied in the following phase `... value that references another author`.

Reply:
> Authors usually help out other authors with their papers. In that role they are called a collaborator.
Add a column to `authors` to store the `author_no` from the collaborator and add a key for integrity validation.

Is it better like this?

Reply:
Why not remove `add a key for integrity validation` ?
or just explicitly say `add a foreign key` ?

Reply:
The first part is just to tell to add the column that should accomodate values equal to `author_no` instead of the name of the collaborator for example. And I tried to formulate the intend instead of the practical implementation (without telling "why"). 

I'll rephrase it to say "add a foreign key"